src/GraphQL.Analyzers.Tests/FieldBuilderAnalyzerTests.cs

```diff
@@ -0,0 +1,613 @@
+                    graphType.Field<StringGraphType>("name")
+                        .Resolve(context => "text")
+                        .Description("description");
+                }
+            }
+            """;
+
+        var expected = VerifyCS.Diagnostic(FieldBuilderAnalyzer.DoNotUseObsoleteFieldMethods).WithSpan(9, 9, 12, 40);
+        await VerifyCS.VerifyCodeFixAsync(source, expected, fix).ConfigureAwait(false);
+    }
+
+    [Fact]
+    public async Task FieldAsync_FixAsFieldAndResolveAsync()
+    {
+        const string source = """
+            using System.Threading.Tasks;
+            using GraphQL.Types;
+
+            namespace Sample.Server;
+
+            public class MyGraphType : ObjectGraphType
+            {
+                public MyGraphType()
+                {
+                    FieldAsync<StringGraphType>(
+                        "name",
+                        "description",
+                        resolve: context => Task.FromResult<object>("text"));
+                }
+            }
+            """;
+
+        const string fix = """
+            using System.Threading.Tasks;
+            using GraphQL.Types;
+
+            namespace Sample.Server;
+
+            public class MyGraphType : ObjectGraphType
+            {
+                public MyGraphType()
+                {
+                    Field<StringGraphType>("name")
+                        .Description("description")
+                        .ResolveAsync(context => Task.FromResult<object>("text"));
+                }
+            }
+            """;
+
+        var expected = VerifyCS.Diagnostic(FieldBuilderAnalyzer.DoNotUseObsoleteFieldMethods).WithSpan(10, 9, 13, 65);
+        await VerifyCS.VerifyCodeFixAsync(source, expected, fix).ConfigureAwait(false);
+    }
+
+    [Fact]
+    public async Task ArgumentsListMultilineFormatted_FormattingPreserved()
+    {
+        const string source = """
+            using GraphQL.Types;
+
+            namespace Sample.Server;
+
+            public class MyGraphType : ObjectGraphType
+            {
+                public MyGraphType()
+                {
+                    Field<StringGraphType>(
+                        "name",
+                        arguments: new QueryArguments(new QueryArgument<StringGraphType>
+                        {
+                            Name = "argName"
+                        }),
+                        resolve: context => "text");
+                }
+            }
+            """;
+
+        const string fix = """
+            using GraphQL.Types;
+
+            namespace Sample.Server;
+
+            public class MyGraphType : ObjectGraphType
+            {
+                public MyGraphType()
+                {
+                    Field<StringGraphType>("name")
+                        .Arguments(new QueryArguments(new QueryArgument<StringGraphType>
+                        {
+                            Name = "argName"
+                        }))
```

Comment:
Have you considered an analyzer to convert arguments to the new syntax also?  I'm thinking probably a separate analyzer.  It would convert this old syntax to this:

```cs
                        .Argument<StringGraphType>("argName")
                        // additional arguments on separate lines
```

Reply:
Yeah, I thought about it but didn't implement it yet because I didn't need it in our code base. But I can do it before we merge this PR.

Reply:
I think I'll postpone this to a separate PR. Since the old syntax is not deprecated, this should probably be implemented as a `CodeRefactoringProvider` rather than `CodeFixProvider`. The difference is that refactoring isn't backed by diagnostic rules and doesn't produce compiler warnings. It will appear under the "Quick Actions" menu similar to this
![image](https://github.com/graphql-dotnet/graphql-dotnet/assets/4518483/e59a1a72-3742-42b3-a511-0ebb8db53558)

Reply:
> I think I'll postpone this to a separate PR. Since the old syntax is not deprecated, this should probably be implemented as a `CodeRefactoringProvider` rather than `CodeFixProvider`.

Sounds good.